### PR TITLE
arm64: dts: qcom: samsung-j5nlte: add display support

### PR DESCRIPTION
**Pull request by nergzd723**

### CODE REVIEW

File: arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts [hunk 1]
```diff
@@ -10,7 +10,6 @@
 / {
 	model = "Samsung Galaxy J5 2015 (SM-J500FN)";
 	compatible = "samsung,j5nlte", "qcom,msm8916";
-
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts:13]: revert

**Comment by nergzd723** [comment]
> There are various checkpatch complains here (see CI log). Linux kernel uses tabs for indentation.

I thought I've used tabs, but my beloved Visual Studio Code replaced them with spaces

File: arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts (hunk 1 above)
Comment (nergzd723) [reply arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts]: done

File: arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts [hunk 2]
```diff
@@ -115,6 +129,72 @@
 	};
 };
 
+&dsi0 {
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&mdss_default>;
+	pinctrl-1 = <&mdss_sleep>;
+		panel@0 {
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts:136]: the indentation looks still weird here...

File: arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts [hunk 3]
```diff
@@ -115,6 +129,72 @@
...
+				reg = <0>;
+
+				compatible = "samsung,s6e8aa5x01-ams497hy01";
+
+				vdd3-supply = <&pm8916_l17>;
+				vci-supply = <&pm8916_l6>;
+				power-supply = <&pm8916_l17>;
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts:143]: l17 is already listed as vdd3-supply

File: arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts [hunk 4]
```diff
@@ -115,6 +129,72 @@
...
+
+				reset-gpios = <&msmgpio 25 GPIO_ACTIVE_HIGH>;
+
+				ports {
+					#address-cells = <1>;
+					#size-cells = <0>;
+					port@0{
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts:150]:
```suggestion
					port@0 {
```

File: arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts [hunk 5]
```diff
@@ -115,6 +129,72 @@
...
+		};
+
+		ports {
+				port@1 {
+						dsi0_out: endpoint {
+								remote-endpoint = <&panel_in>;
+								data-lanes = <0 1 2 3>;
+						};
+				};
+		};
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts:166]: very weird indentation here as well (two tabs?)

File: arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts [hunk 6]
```diff
@@ -115,6 +129,72 @@
...
+};
+
+&msmgpio {
+		pmx_mdss {
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts:170]:
```suggestion
	pmx_mdss {
```

File: drivers/gpu/drm/panel/panel-simple.c [hunk 7]
```diff
@@ -2935,7 +2935,7 @@ static const struct panel_desc sharp_lq070y3dg3b = {
 	},
 	.bus_format = MEDIA_BUS_FMT_RGB888_1X24,
 	.bus_flags = DRM_BUS_FLAG_DE_HIGH | DRM_BUS_FLAG_PIXDATA_POSEDGE |
-		     DRM_BUS_FLAG_SYNC_DRIVE_POSEDGE,
+			 DRM_BUS_FLAG_SYNC_DRIVE_POSEDGE,
```
Comment (Minecrell) [inline drivers/gpu/drm/panel/panel-simple.c:2938]: All these changes :sob:

File: drivers/gpu/drm/panel/panel-simple.c [hunk 8]
```diff
@@ -3815,6 +3815,36 @@ struct panel_desc_dsi {
...
+				.num_modes = 1,
+				.bpc = 8,
+				.size = {
+						.width = 62,
+						.height = 106,
+				},
+		},
+		.flags = MIPI_DSI_MODE_VIDEO | MIPI_DSI_MODE_VIDEO_BURST,
+		.format = MIPI_DSI_FMT_RGB888,
+		.lanes = 4,
```
Comment (Minecrell) [inline drivers/gpu/drm/panel/panel-simple.c:3845]: two tabs again :3

File: drivers/gpu/drm/panel/panel-simple.c (hunk 7 above)
Comment (nergzd723) [reply drivers/gpu/drm/panel/panel-simple.c:2938]: I've replaced all spaces with tabs... oops

**Comment by nergzd723** [comment]
Fixed, but is interesting to me how I managed to indent with two tabs

File: arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts [hunk 9]
```diff
@@ -115,6 +129,71 @@
 	};
 };
 
+&dsi0 {
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&mdss_default>;
+	pinctrl-1 = <&mdss_sleep>;
+	panel@0 {
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts:136]: Can you add an empty line before this for consistency?

Comment (nergzd723) [reply arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts:137]: fixed... probably

File: arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts [hunk 10]
```diff
@@ -115,6 +129,72 @@
...
+		};
+	};
+
+	ports {
+		port@1 {
+			dsi0_out: endpoint {
+					remote-endpoint = <&panel_in>;
+					data-lanes = <0 1 2 3>;
+				};
+			};
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts:165]: the indentation is messed up here

File: arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts [hunk 11]
```diff
@@ -115,6 +129,72 @@
...
+	pinctrl-0 = <&mdss_default>;
+	pinctrl-1 = <&mdss_sleep>;
+
+	panel@0 {
+		reg = <0>;
+
+		compatible = "samsung,s6e8aa5x01-ams497hy01";
+
+		vdd3-supply = <&pm8916_l17>;
+		vci-supply = <&pm8916_l6>;
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts:143]: AFAIK vci is usually the 3V regulator and vdd3 ~1.8V. It sounds confusing, but for some reason it's like that.

so vdd3 should be l6 and vci l17.

Comment (nergzd723) [reply arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts]: done

File: arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts (hunk 10 above)
Comment (nergzd723) [reply arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts:164]: You mean brackets? If so, I've fixed them

**Comment by Minecrell** [comment]
Fixed one more indentation problem and merged it. Thank you!

**Pull request merged**